addons/hr_recruitment/models/hr_job.py

```diff
@@ -108,7 +108,15 @@ def _alias_get_creation_values(self):
     @api.model
     def create(self, vals):
         vals['favorite_user_ids'] = vals.get('favorite_user_ids', []) + [(4, self.env.uid)]
-        return super(Job, self).create(vals)
+        new_job = super(Job, self).create(vals)
+        utm_linkedin = self.env['utm.source'].search([["name", "ilike", "linkedin"]])
+        if not self.env['hr.recruitment.source'].search_count([["source_id", "=", utm_linkedin.id], ["job_id", "=", new_job.id]]):
```

Comment:
I don't know if this check is useful as you are creating the job and there is no code that automatically creates a 'hr.recruitment.job'. Instead, you can check that the utm_linkedin exists.